server.js

```diff
@@ -17,93 +17,124 @@ client.on('error', err => console.error(err));
 
 // ////// call the handler function 
 
-server.get('/location',locationHandler);
+server.get('/location', locationHandler);
 server.get('/weather', weatherHanddler);
 server.get('/events', eventHanddler); ////// here i have a problem and gorob help me to solve it //////
 
 // /////// location handler //////
-  function locationHandler(req,res){
-getlocation(req.query.data)
-.then(locationData => res.status(200).json(locationData));
-  };
+function locationHandler(req, res) {
+  const city = req.query.data;
+  getlocation(city)
+    // .then(locationData => res.status(200).json(locationData));
+    .then(data => sendJson(data, res))
```

Comment:
You are calling a function clled `sendJson()` here.  That's from the solution code ... 

- What do you think it's supposed to do? 
- Why does it have those arguments?
- You haven't actually written it ... so I wonder if you write it, will your code work better?